creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1004,6 +927,103 @@ class SnapbackSM {
     return clockValue
   }
 
+  /*
+   * For every node user, record sync requests to issue to secondaries if this node is primary
+   *    and record replica set updates to issue for any unhealthy replicas
+   *
+   * Purpose for the if/else case is that if the current node is a primary, issue reconfig or sync requests.
+   * Else, if the current node is a secondary, only issue reconfig requests.
+   *
+   * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
+   */
+  async aggregateReconfigAndPotentialSyncOps (nodeUsers, unhealthyPeers) {
+    const requiredUpdateReplicaSetOps = []
+    const potentialSyncRequests = []
+
+    for (const nodeUser of nodeUsers) {
+      const { primary, secondary1, secondary2, primarySpID, secondary1SpID, secondary2SpID } = nodeUser
+
+      let unhealthyReplicas = []
+
+      /**
+       * If this node is primary for user, check both secondaries for health
+       * Enqueue SyncRequests against healthy secondaries, and enqueue UpdateReplicaSetOps against unhealthy secondaries
+       */
+      let replicaSetNodesToObserve = [
+        { endpoint: secondary1, spId: secondary1SpID },
+        { endpoint: secondary2, spId: secondary2SpID }
+      ]
+
+      if (primary === this.endpoint) {
+        // filter out false-y values to account for incomplete replica sets
+        replicaSetNodesToObserve = replicaSetNodesToObserve.filter(entry => entry.endpoint && entry.spId)
```

Comment:
i think this `entry.spID` non-null check is unnecessary

Reply:
yeah, I think we can remove this spId check actually. removing